app/src/main/res/layout/adapter_kind_items.xml

```diff
@@ -14,56 +14,78 @@
             android:textStyle="bold"
-            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintBottom_toBottomOf="@+id/stallImage"
             app:layout_constraintEnd_toEndOf="@+id/stallImage"
-            app:layout_constraintStart_toStartOf="@+id/stallImage"
+            app:layout_constraintStart_toEndOf="@+id/kImg"
             app:layout_constraintTop_toTopOf="@+id/stallImage" />
 
     <TextView
             android:id="@+id/price"
-            android:layout_width="30dp"
-            android:layout_height="0dp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="169dp"
+            android:layout_marginTop="56dp"
+            android:layout_marginBottom="16dp"
             android:fontFamily="@font/shellypalmer_circular_std_book"
             android:gravity="center"
             android:text="0"
             android:textAlignment="center"
             android:textColor="@color/colorWhite"
-            android:textSize="24sp"
-            android:textStyle="bold"
-            app:layout_constraintBottom_toBottomOf="@+id/ItemName"
-            app:layout_constraintEnd_toEndOf="@+id/stallImage"
-            app:layout_constraintHorizontal_bias="0.6"
-            app:layout_constraintStart_toEndOf="@+id/ItemName"
-            app:layout_constraintTop_toTopOf="@+id/ItemName" />
+            android:textSize="22sp"
+            app:layout_constraintBottom_toBottomOf="@+id/stallImage"
+            app:layout_constraintStart_toStartOf="@+id/stallImage"
+            app:layout_constraintTop_toTopOf="@+id/stallImage" />
 
     <ImageView
             android:id="@+id/kImg"
-            android:layout_width="56dp"
-            android:layout_height="50dp"
-            app:layout_constraintBottom_toBottomOf="@+id/ItemName"
-            app:layout_constraintEnd_toStartOf="@+id/ItemName"
-            app:layout_constraintHorizontal_bias="0.28"
+            android:layout_width="49dp"
+            android:layout_height="56dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="25dp"
+            android:layout_marginBottom="25dp"
+            app:layout_constraintBottom_toBottomOf="@+id/stallImage"
             app:layout_constraintStart_toStartOf="@+id/stallImage"
-            app:layout_constraintTop_toTopOf="@+id/ItemName"
+            app:layout_constraintTop_toTopOf="@+id/stallImage"
             app:srcCompat="@drawable/ic_card_giftcard_black_24dp" />
 
+    <TextView
+            android:id="@+id/textView31"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="52dp"
+            android:layout_marginBottom="13dp"
+            android:fontFamily="@font/shellypalmer_circular_std_book"
+            android:gravity="center"
+            android:text="Tokens"
```

Comment:
Shift this text to the strings resource file

Reply:
I have made the suggested changes.